demos/dashboards-investor-preferences/demo.js

```diff
@@ -0,0 +1,395 @@
+        // alignment or SFDR classification
+        if (euTaxAlignment.value) {
+            const minPortion = document.getElementById('min-portion').value,
+                isOptional = document.getElementById('optional');
+
+            currentFilter.textContent = 'EU Tax alignment: ' +
+                euTaxAlignment.selectedOptions[0].textContent +
+                ', Minimum portion: ' + minPortion + '%, ' +
+                isOptional.selectedOptions[0].textContent + '.';
+
+            filters.push(
+                {
+                    dataPointId: euTaxAlignment.value,
+                    comparatorCode: 'IN',
+                    value: isOptional.value
+                },
+                {
+                    dataPointId: 'EET_PlannedInvtsSustInvEUTaxonomy',
+                    comparatorCode: 'GT',
+                    value: minPortion
+                }
+            );
+        }
+        if(sfdrTypes.length) {
+            const values = [],
+                types = [];
+
+            sfdrTypes.forEach(type => {
+                values.push(type.value);
+                types.push(
+                    document.querySelector(`label[for='${type.id}'`).textContent
+                );
+            });
+            currentFilter.textContent +=
+                'SFDR Classification: ' + types.join(', ');
+
+            filters.push({
+                dataPointId: 'EET_EUSFDRType',
+                comparatorCode: sfdrTypes.length > 1 ? 'IN' : 'EQ',
+                value: values.join(':')
+            });
+        }
+
+        setFilter(filters);
+    });
+
+    document.getElementById('prev').addEventListener('click', () => {
+        changePage(false);
+    });
+
+    document.getElementById('next').addEventListener('click', () => {
+        changePage();
+    });
+
+}
+
+async function handleSelectEnvironment (evt) {
+    const target = evt.target;
+    const postmanJSON = await getPostmanJSON(target);
+    const filters = document.querySelectorAll(
+        '.input-wrapper input, .input-wrapper select, button#filter-1'
+    );
+
+    if (!postmanJSON) {
+        loadingLabel.textContent =
+            'The provided file is not a Postman Environment Configuration.';
+        loadingLabel.style.display = 'block';
+
+        return;
+    }
+
+    for (const filter of filters) {
+        filter.disabled = false;
+    }
+
+    target.parentNode.style.display = 'none';
+
+    loadingLabel.style.display = 'block';
+    loadingLabel.textContent = 'Loading data…';
+
+    displayInvestorPreferences(postmanJSON);
+}
+
+document
+    .getElementById('postman-json')
+    .addEventListener('change', handleSelectEnvironment);
+
+async function getPostmanJSON (htmlInputFile) {
+    let file;
+    let fileJSON;
+
+    for (file of htmlInputFile.files) {
+        try {
+            fileJSON = JSON.parse(await file.text());
+            if (HighchartsConnectors.Morningstar.isPostmanEnvironmentJSON(fileJSON)) {
```

Comment:
It gives a warning in the console. This function is found here nowadays:
```suggestion
            if (HighchartsConnectors.Morningstar.Shared.isPostmanEnvironmentJSON(fileJSON)) {
```